src/config/verification.rs

```diff
@@ -170,3 +183,42 @@ pub(crate) async fn build_sigstore_trust_root() -> Result<Option<Arc<SigstoreTru
         sigstore::trust::sigstore::SigstoreTrustRoot::new(Some(checkout_path.as_path())).await?;
     Ok(Some(Arc::new(trust_root)))
 }
+
+fn build_sigstore_trust_root_from_config(
+    pki_file: &PathBuf,
+) -> Result<Option<Arc<SigstoreTrustRoot>>> {
+    debug!(
+        "Using user specified Sigstore trust root location: {}",
+        pki_file.display()
+    );
+    let json_bytes = fs::read(pki_file).map_err(|e| {
+        anyhow!(
+            "could not read Sigstore PKI file {}: {:?}",
+            pki_file.display(),
+            e
+        )
+    })?;
+    // Sigstore.rs does not have a direct way to read from file, so we parse the
+    // protobuf spec first, then re-serialize the trusted_root field to JSON bytes
+    let client_trust_config: sigstore_protobuf_specs::dev::sigstore::trustroot::v1::ClientTrustConfig =
```

Comment:
I would prefer to have this import on the top of the file, but not a blocker for me.